rust/crd/src/ldap.rs

```diff
@@ -0,0 +1,216 @@
+                return Some(DruidLdapSettings {
+                    ldap: provider.clone(),
+                });
+            }
+        }
+        None
+    }
+
+    fn add_druid_system_authenticator_config(&self, config: &mut BTreeMap<String, Option<String>>) {
+        const PREFIX: &str = "druid.auth.authenticator.DruidSystemAuthenticator";
+
+        config.insert(format!("{PREFIX}.type"), Some("basic".to_string()));
+        config.insert(
+            format!("{PREFIX}.credentialsValidator.type"),
+            Some("metadata".to_string()),
+        );
+
+        // this line is left out, as we don't want to create an admin user
+        // # druid.auth.authenticator.DruidSystemAuthenticator.initialAdminPassword: XXX
+
+        config.insert(
+            format!("{PREFIX}.initialInternalClientPassword"),
+            Some(PLACEHOLDER_INTERNAL_CLIENT_PASSWORD.to_string()),
+        );
+        config.insert(
+            format!("{PREFIX}.authorizerName"),
+            Some("DruidSystemAuthorizer".to_string()),
+        );
+        config.insert(format!("{PREFIX}.skipOnFailure"), Some("true".to_string()));
+    }
+
+    fn add_ldap_authenticator_config(&self, config: &mut BTreeMap<String, Option<String>>) {
+        const PREFIX: &str = "druid.auth.authenticator.Ldap";
+
+        config.insert(format!("{PREFIX}.type"), Some("basic".to_string()));
+        config.insert(
+            format!("{PREFIX}.enableCacheNotifications"),
+            Some("true".to_string()),
+        );
+        config.insert(
+            format!("{PREFIX}.credentialsValidator.type"),
+            Some("ldap".to_string()),
+        );
+        config.insert(
+            format!("{PREFIX}.credentialsValidator.url"),
+            Some(self.credentials_validator_url()),
+        );
+
+        // we only add these lines if bind credentials are configured
+        if self.ldap.bind_credentials.is_some() {
+            config.insert(
+                format!("{PREFIX}.credentialsValidator.bindUser"),
+                Some(PLACEHOLDER_LDAP_BIND_USER.to_string()), // NOTE: this placeholder will be replaced from a mounted secret on container startup
+            );
+            config.insert(
+                format!("{PREFIX}.credentialsValidator.bindPassword"),
+                Some(PLACEHOLDER_LDAP_BIND_PASSWORD.to_string()), // NOTE: this placeholder will be replaced from a mounted secret on container startup
+            );
+        } else {
+            // WIP: attempt to use LDAP without bind credentials.
+            // Setting an empty password and user does not work.
+            // Not-specifying these fields at all leads to null pointer exceptions.
+            config.insert(
+                format!("{PREFIX}.credentialsValidator.bindUser"),
+                Some("".to_string()),
+            );
+            config.insert(
+                format!("{PREFIX}.credentialsValidator.bindPassword"),
+                Some("".to_string()),
+            );
+        }
```

Comment:
Since this does not work, i would log an error in the else part, add a TODO linking to the "research anonymous ldap" ticket.